### PR TITLE
Cleanup v2

### PR DESCRIPTION
**Pull request by helo9**
Concurrent PR for #3

### CODE REVIEW

File: src/d39_bt/d39_bt.c [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+	if (err) {
+		printk("Advertising failed to start (err %d)\n", err);
+		return;
+	}
+
+	printk("Connection Callbacks registered\n");
+}
+
+int d39_bt_enable() {
+	return bt_enable(bt_ready);
```
Comment (dernatsch) [inline src/d39_bt/d39_bt.c:73]: I don't quite get the API here.
`bt_enable` is given a function pointer *(void)(int), right?
If bt_ready now fails, does bt_enable also return the error code that bt_ready gives?

If so I think the error handling (printk) should be moved out of bt_ready. If not I need to think some more about it.
Any way I think it's not ideal that the handling is in bt_ready and not in the main loop.

File: src/d39_bt/d39_bt.c [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+}
+
+void d39_bt_send_msg(char *buf, size_t len) {
```
Comment (dernatsch) [inline src/d39_bt/d39_bt.c:76]: int d39_bt...

File: src/config.h [hunk 3]
```diff
@@ -0,0 +1,9 @@
+/* GPIO */
```
Comment (dernatsch) [inline src/config.h:1]: Wouldn't hurt to add a `#pragma once` here

File: src/d39_gpio/d39_gpio.c [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+atomic_t interrupt_count = ATOMIC_INIT(0);
+
+void handle_tick(const struct device *dev, struct gpio_callback *cb, uint32_t pint) {
+    atomic_inc(&interrupt_count);
+}
+
+/**
+ * Sets up the gpio pin and adds the corresponding callback.
+ * @param err Gets set to an error message in case of an error.
```
Comment (dernatsch) [inline src/d39_gpio/d39_gpio.c:19]: Out of date comments

File: src/d39_gpio/d39_gpio.c [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    // These do not return an error code
+    gpio_init_callback(&tick_cb_data, handle_tick, BIT(SENS_PIN));
+    gpio_add_callback(gpio0, &tick_cb_data);
+
+    return 0;
+}
+
+/**
+ * Gets the interrupt count and sets it to zero.
+ * @param err Gets set to a error message in case of an error.
```
Comment (dernatsch) [inline src/d39_gpio/d39_gpio.c:47]: Out of date

File: src/main.c [hunk 6]
```diff
@@ -112,109 +47,13 @@ void main(void)
...
-	}
-
-	printk("Disconnected (reason %u)", reason);
-}
-
-static void ccc_cfg_changed(const struct bt_gatt_attr *attr, uint16_t value)
-{
-	ARG_UNUSED(attr);
-	notify_enable = (value == BT_GATT_CCC_NOTIFY);
+	d39_bt_send_msg( &buf, 15);
```
Comment (dernatsch) [inline src/main.c:58]: random space before &

**Review by dernatsch — CHANGES_REQUESTED** [review]

File: src/d39_bt/d39_bt.c (hunk 1 above)
Comment (helo9) [reply src/d39_bt/d39_bt.c:73]: From my understanding no, because bt initialization might take to long. therefore bt_enable is non blocking

Comment (dernatsch) [reply src/d39_bt/d39_bt.c:73]: Okay, so if the initialization in bt_enable fails, we don't get any feedback and don't try again.
How about we let bt_enable send a signal if it succeeds or fails and have the main thread wait for it?

Then we could retry on a fail

Comment (helo9) [reply src/d39_bt/d39_bt.c:73]: Will take a look at other examples.

**Comment by dernatsch** [comment]
Closes #3

File: src/main.c (hunk 6 above)
Comment (helo9) [reply src/main.c:58]: fixed

File: src/d39_bt/d39_bt.c (hunk 2 above)
Comment (helo9) [reply src/d39_bt/d39_bt.c:76]: wowowowow, fixed

File: src/d39_gpio/d39_gpio.c (hunk 4 above)
Comment (helo9) [reply src/d39_gpio/d39_gpio.c:19]: removed

File: src/d39_gpio/d39_gpio.c (hunk 5 above)
Comment (helo9) [reply src/d39_gpio/d39_gpio.c:47]: removed

File: src/config.h (hunk 3 above)
Comment (helo9) [reply src/config.h:1]: sure, added

File: src/d39_bt/d39_bt.c (hunk 1 above)
Comment (helo9) [reply src/d39_bt/d39_bt.c:73]: Have some idea, how to do it, but would do this in anther PR and merge it now.

**Pull request merged**